analysis/estimate/notebooks/run-estimate.Rmd

```diff
@@ -0,0 +1,289 @@
+
+Second, what do we see for stromal scores?
+
+* For SCPCP000001 and SCPCP000002, pseudobulk scores look either on par with bulk or _maybe_ a bit lower, but that is mostly in SCPCP000001 and one outlier in SCPCP000002
+* For the other three projects, pseudobulk and bulk scores appear highly similar
+
+
+
+The bulk distributions appear very similar to one another, so let's compare them directly next:
+
+
+
+```{r fig.width=10}
+expr_wide_df <- estimate_df |>
+  tidyr::pivot_wider(
+    names_from = expression_type, 
+    values_from = score
+  )
+
+ggplot(expr_wide_df) + 
+  aes(x = bulk_counts, y = bulk_tpm) + 
+  geom_point() +
+  geom_abline(color = "darkorange2") +
+  coord_equal() +
+  facet_grid(
+    rows = vars(score_type), 
+    cols = vars(project_id)
+  ) 
+```
+
+Bulk counts do not take gene length into account, but TPM does. 
+Across all projects, we see that immune scores tend to be slightly higher for TPM, and by contrast stromal scores tend to be slightly lower for TPM compared to bulk.
+The effect sizes are fairly small, but the effects seem likely to be significant, which we'll check separately for `stromal` and `immune`.
+Below, we fit regressions that fix the slope to 1, so the intercept will tell us the bias from `y=x`:
+
+```{r}
+stromal_df <- expr_wide_df |>
+  dplyr::filter(score_type == "stromal")
+summary(lm(bulk_tpm ~ 1 + offset(bulk_counts), data = stromal_df))
+
+
+immune_df <- expr_wide_df |>
+  dplyr::filter(score_type == "immune")
+summary(lm(bulk_tpm ~ 1 + offset(bulk_counts), data = immune_df))
+
+```
+
+Indeed, effect sizes are not very large in the scheme of the full range of values, but they are highly significant: Compared to bulk counts, immune scores are relatively higher when calculated with TPM, and stromal scores are relatively lower whem calculated with TPM.
+
+Let's directly compare bulk to and pseudobulk.
+Since TPM and bulk counts were very similar, we'll just visualize with the bulk _counts_ (not TPM) here:
+
+
+```{r fig.width=10}
+compare_modalities_df <- estimate_df |>
+  tidyr::pivot_wider(
+    names_from = expression_type, 
+    values_from = score
+  ) 
+
+ggplot(compare_modalities_df) + 
+  aes(x = bulk_counts, y = pseudobulk) + 
+  geom_point() +
+  geom_abline(color = "darkorange2") +
+  geom_smooth(method = "lm") +
+  facet_grid(
+    rows = vars(score_type), 
+    cols = vars(project_id)
+  ) +
+  labs(title ="pseudobulk ~ bulk counts") +
+  theme(aspect.ratio = 1)
+```
+
+* There does not appear to be a significant relationship between modalities for SCPCP000002 and SCPCP000006, for either score 
+* For SCPCP000001, immune scores tend to be higher in pseudobulk but stromal scores are about the same between modalities
+* For SCPCP000006 and SCPCP0000017, immune scores tend to be lower in pseudobulk but stromal scores are about the same between modalities
+* Note that projects SCPCP000001 and SCPCP000002 are single-cell, but the other three projects are single-nucleus
```

Comment:
It might be worth a plot where you just look at immune and stromal scores in pseudobulk and group by `seq_unit`